client/asset/eth/nodeclient_harness_test.go

```diff
@@ -143,6 +139,33 @@ func newRedeem(secret, secretHash [32]byte) *asset.Redemption {
 	}
 }
 
+// waitForReceipt waits for a tx. This is useful on testnet when a tx may be "missing"
+// due to reorg. Wait for a few blocks to find the main chain and hopefully our tx.
+func waitForReceipt(t *testing.T, nc *nodeClient, hash common.Hash) (*types.Receipt, error) {
+	t.Helper()
+	// Waiting as much as five blocks.
+	timesUp := time.After(time.Duration(5*secPerBlock) * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		case <-time.After(time.Second):
+			receipt, err := nc.transactionReceipt(ctx, hash)
+			if err != nil {
+				if errors.Is(err, asset.CoinNotFoundError) {
+					fmt.Println("got um error", err.Error())
+					continue
+				}
+				return nil, err
+			}
+			spew.Dump(receipt)
+			return receipt, nil
+		case <-timesUp:
+			return nil, errors.New("wait for receipt timed out")
```

Comment:
Like you've mentioned [here](https://github.com/decred/dcrdex/issues/1687#issuecomment-1169693575), low gas price could indeed be the reason for txn not getting mined (or maybe even outright not accepted by other nodes),

maybe mention this in error message to remind checking that hypothesis (print out gas price value used, and suggest checking it against current rough value for gas price in blockchain explorer) when test fails ? also, might extend error message with something like `txn might be missing due to reorg`